api/enki.py

```diff
@@ -0,0 +1,539 @@
+    log = logging.getLogger("Enki Bibliographic Parser")
+    log.setLevel(logging.DEBUG)
+
+    def __init__(self, include_availability=True, include_bibliographic=True):
+        self.include_availability = include_availability
+        self.include_bibliographic = include_bibliographic
+
+    def extract_bibliographic(self, element):
+        identifiers = []
+        contributors = []
+        identifiers.append(IdentifierData(Identifier.ISBN, element["isbn"]))
+        sort_name = element["author"]
+        if not sort_name:
+            sort_name = Edition.UNKNOWN_AUTHOR
+        contributors.append(ContributorData(sort_name=sort_name))
+        primary_identifier = IdentifierData(EnkiAPI.ENKI_ID, element["id"])
+        metadata = Metadata(
+        data_source=DataSource.ENKI,
+        title=element["title"],
+        language="eng",
+        medium=Edition.BOOK_MEDIUM,
+        #series=series,
+        publisher=element["publisher"],
+        #imprint=imprint,
+        #published=publication_date,
+        primary_identifier=primary_identifier,
+        identifiers=identifiers,
+        #subjects=subjects,
+        contributors=contributors,
+        )
+        licenses_owned=element["availability"]["totalCopies"]
+        licenses_available=element["availability"]["availableCopies"]
+        hold=element["availability"]["onHold"]
+        formats = []
+        formats.append(FormatData(content_type=Representation.EPUB_MEDIA_TYPE, drm_scheme=DeliveryMechanism.ADOBE_DRM))
+
+        circulationdata = CirculationData(
+            data_source=DataSource.ENKI,
+            primary_identifier=primary_identifier,
+            formats=formats,
+            licenses_owned = int(licenses_owned),
+            licenses_available = int(licenses_available),
+            patrons_in_hold_queue = int(hold)
+        )
+
+        metadata.circulation = circulationdata
+        return metadata
+
+
+    def process_one(self, element):
+        if self.include_bibliographic:
+            bibliographic = self.extract_bibliographic(element)
+        else:
+            bibliographic = None
+
+        availability = None
+        if self.include_availability:
+            availability = bibliographic.circulation
+
+        return bibliographic, availability
+
+class EnkiImport(CollectionMonitor):
+    """Import content from Enki that we don't yet have in our collection
+    """
+    SERVICE_NAME = "Enki Circulation Monitor"
+    INTERVAL_SECONDS = 500
+    PROTOCOL = EnkiAPI.ENKI_EXTERNAL
+    DEFAULT_BATCH_SIZE = 100 
+    FIVE_MINUTES = datetime.timedelta(minutes=5)
+    
+    def __init__(self, _db, collection, api_class=EnkiAPI):
+        """Constructor."""
+        super(EnkiImport, self).__init__(_db, collection)
+        self.api = api_class(_db, collection)
+        self.analytics = Analytics(_db)
+        self.bibliographic_coverage_provider = (
+            EnkiBibliographicCoverageProvider(_db, collection, api_class=self.api)
```

Comment:
Ah, yes, apparently you'd be right. We test creating license pools, but not via EnkiImport. That'll be something to add.